docs/runbook.md

```diff
@@ -13,103 +13,195 @@
+        - [nodes](#nodes)
+        - [nodes_requirement](#nodes_requirement)
+          - [type](#type)
   - [platform](#platform)
   - [testcase](#testcase)
+    - [criteria](#criteria)
 
 ## How-to
 
 ### Configure Azure deployment
 
+Start with lisa\microsoft\runbook\azure.yml, refer this sample yml file to deploy nodes on Azure.
+
 ### Select and set test cases
 
+[testcase](#testcase) section is used to select test cases, we can select the cases by specify 'area', 'category', 'priority', 'tags' in criteria, specify 'include', 'exclude', 'forceInclude', 'forceExclude' for select action.
+
 ### Use variable and secrets
 
+Using variable and secrets is to avoid updating many places for the same value and pass variable value from command line, keep sensitive data shown with mask.
+
 ### Use partial runbook
 
+With [parent](#parent) section, we can reuse the parent yml file which contains the common use definition.
 
 ### Use extensions
 
+With [extension](#extension) section, we can load test cases and modules, this helps us category test cases and modules based on its usage.
+
 ## Reference
 
 ### name
 
+type: str, optional, default is "not_named"
+
+Set the name of the test run. This name will be used to group results and put it in title, also the created resources's name contains this specified str.
+
 ### test_project
 
+type: str, optional, default is empty
+
+Set the project name of this test run. This name will be used to group test results in html.
+
 ### test_pass
 
+type: str, optional, default is empty
+
+Set the test pass name of this test run. This name combined with test project will be used to group test results in html.
+
 ### tags
 
+type: list of str, optional, default is empty
+
+Set the tags of the test run. This name combined with test project and test pass names will be used to group test results in html.
+
 ### parent
 
+Share the runbook for similar runs.
+
 #### path
 
-#### strategy
+type: str, optional, default is empty
 
-Not implemented
+Specify the path of the parent runbook.
 
 ### extension
 
+type: list of str, optional, default: empty
+
+Specify the path of the LISA module for loading test cases and extensions.
+
 ### variable
 
+Used to support variables in other fields.
+
 #### is_secret
 
-#### file
+type: boolean, optional, default is False.
 
-#### name
+If the variable is secret, the value of this variable will be removed from log and other output information.
 
-### value
+#### file
 
-### artifact
+type: list of str, optional, default: empty
+
+Specify path of other yml files which define variables.
 
 #### name
 
-#### locations
+type: str, optional, default is empty.
 
-##### type
+Variable name.
 
-##### path
+#### value
+
+type: str, optional, default is empty
 
+Value of the paired variable.
 
 ### notifier
 
+Receive messages during the test run and output them somewhere. The supported messages are derived from `MessageBase`, including `TestRunMessage`, `TestResultMessage`, etc.
```

Comment:
This is usage introduction, it doesn't need to introduce the classes of code. It can have a link to extension to help users, if they want to extend a notifier.